### ISSUE TITLE
Be able to use mosaic without viewlet and portlet managers

### ISSUE THREAD

**Issue by vangheem** [plone/plone.app.mosaic#211]
# Problem

Right now, plone.app.standardtiles provides glue between all the logic we have in the viewlet machinery and makes it available in tiles. It should be possible to provide plone, powered by mosaic, without any portlet or viewlet packages installed. This would probably mean merging plone.app.layout and plone.app.standardtiles.

See also https://github.com/plone/plone.app.mosaic/issues/210

This is a long term goal--not easy to do right now.

**Comment by datakurre** [comment]
FWIW, originally plone.app.standardtiles did implement all viewlets without viewlet, but because that was done mostly by copying code from plone.app.layout and nobody maintaining it, they got eaten by bit rot. Really not sure, how to do that in a maintainable way.

**Comment by jensens** [comment]
I think its a good idea to have the generic logic+template for the viewlet and tile at the same place. its in fact very very similar, just the registration is different.

**Comment by datakurre** [comment]
Agreed. It may not need much that the same classes could be registered both as viewlets and tiles.
